pkg/services/ngalert/api/api.go

```diff
@@ -49,6 +50,27 @@ type Alertmanager interface {
 	TestReceivers(ctx context.Context, c apimodels.TestReceiversConfigBodyParams) (*notifier.TestReceiversResult, error)
 }
 
+type MultiOrgAlertmanager interface {
```

Comment:
The API layer logic previously depended directly on the concrete `Alertmanager` and `MultiOrgAlertmanager` classes. I refactored this to depend on an abstraction of functionality instead. This decouples the code, makes the API handlers unit-testable, and provides slightly better demarcation between the layers.